libraries/c_sdk/standard/ble/test/iot_mqtt_ble_system_test.c

```diff
@@ -0,0 +1,1034 @@
+/*
+ * FreeRTOS
+ * Copyright (C) 2020 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/**
+ * @file iot_mqtt_ble_system_test.c
+ * @brief Integration tests for the coreMQTT library for the FreeRTOS platform
+ * over BLE transport interface.
+ */
+
+/* Standard header includes. */
+#include <string.h>
+#include <stdbool.h>
+#include <assert.h>
+#include <stdlib.h>
+#include <time.h>
+
+#include "FreeRTOS.h"
+
+/* Include task.h for delay function. */
+#include "task.h"
+
+/* Include sempahore for callback synchronization. */
+#include "semphr.h"
+
+/* Unity Test framework include. */
+#include "unity_fixture.h"
+
+/* Include connection configurations header. */
+#include "aws_clientcredential.h"
+
+/* Include paths for public enums, structures, and macros. */
+#include "core_mqtt.h"
+#include "core_mqtt_state.h"
+
+#include "iot_ble.h"
+
+/* Include BLE implementation of transport interface. */
+#include "iot_ble_mqtt_transport.h"
+
+
+
+/**************************************************/
```

Comment:
Why not put logging config in a separate file similar to other places?

Reply:
This config is used only within this test and what i see in the config files is MQTT library level logging config which is common for all tests. Should we keep this config private to this test to reduce clutter.